### PR TITLE
[DM] [SASS-568] Added in DES connector and call to create or update g…

### PR DESCRIPTION
**Pull request by TheDeadlyPianist**
…iftaid

### Description
Please include a brief summary of the change you've made and the issue that it fixes.

Add a link to the relevant story in Jira
[SASS-568](https://jira.tools.tax.service.gov.uk/browse/SASS-568)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [x]  Have you checked for merge conflicts?
- [x]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [x]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you addressed warnings where appropriate?

##### After PRs been Reviewed
- [ ]  Have you checked the PR Builder passes?
- [ ]  Have you checked for merge conflicts?
- [ ]  Have you checked code coverage isn’t lower than previously?

### CODE REVIEW

File: app/connectors/GiftAidSubmissionConnector.scala [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  def submit(
+              nino: String, taxYear: Int, submissionModel: GiftAidSubmissionModel
+            )(headerCarrier: HeaderCarrier): Future[GiftAidSubmissionResponse] = {
+    implicit val desHC: HeaderCarrier = desHeaderCarrier(headerCarrier)
+
+    val desCall: String = appConfig.desBaseUrl + s"/income-tax/nino/$nino/income-source/charity/" +
+      s"annual/$taxYear"
+
+    val body = Json.obj(
+      "giftAidPayments" -> Json.toJson(submissionModel)
```
Comment (BenCosfordHMRC) [inline app/connectors/GiftAidSubmissionConnector.scala:42]: I think the json can just be the model to json

File: app/models/APIErrorModel.scala [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models
+
+case class APIErrorModel()
```
Comment (BenCosfordHMRC) [inline app/models/APIErrorModel.scala:19]: Can this be removed?

File: conf/prod.routes [hunk 3]
```diff
@@ -1,5 +1,5 @@
 # Add all the application routes to the app.routes file
-->         /income-tax-gift-aid        app.Routes
-->         /                           health.Routes
+->         /income-tax           app.Routes
```
Comment (BenCosfordHMRC) [inline conf/prod.routes:2]: sorry the url wasn't in the ticket - think this is fine to be what it was /income-tax-gift-aid

**Review by BenCosfordHMRC — CHANGES_REQUESTED** [review]

File: test/connectors/httpParsers/GiftAidSubmissionHttpParserSpec.scala [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+      }
+
+      "the response status is BAD_REQUEST" in {
+        val errorBody = Json.prettyPrint(Json.obj("code" -> "AWW_MAN", "reason" -> "not again"))
+        val result = read("POST", "/some-url", httpResponse(BAD_REQUEST, errorBody))
+
+        result shouldBe Left(DesErrorModel(BAD_REQUEST, DesErrorBodyModel("AWW_MAN", "not again")))
+      }
+
+      "the response status is FORBIDDEN" in {
```
Comment (robert-webb-hmrc) [inline test/connectors/httpParsers/GiftAidSubmissionHttpParserSpec.scala:82]: While I am a fan of these, I'm not sure if they should more accurately represent the actual errors we may be seeing

**Review by robert-webb-hmrc — CHANGES_REQUESTED** [review]
Looks mostly fine. Please take a look at Ben's previous comments, and can we address the following warns;
Screenshot 2021-04-12 at 12 37 35
Screenshot 2021-04-12 at 12 40 08

File: app/connectors/GiftAidSubmissionConnector.scala (hunk 1 above)
Comment (TheDeadlyPianist) [reply app/connectors/GiftAidSubmissionConnector.scala:42]: DES is expecting it in this format.

**Review by robert-webb-hmrc — APPROVED** [review]

**Pull request merged**